src/vs/editor/common/viewModel/viewModel.ts

```diff
@@ -101,28 +102,54 @@ export class LineBreakData {
 	constructor(
 		public breakOffsets: number[],
 		public breakOffsetsVisibleColumn: number[],
-		public wrappedTextIndentLength: number
+		public wrappedTextIndentLength: number,
+		public injectionTexts: string[] | null,
+		public injectionOffsets: number[] | null
 	) { }
 
-	public static getInputOffsetOfOutputPosition(breakOffsets: number[], outputLineIndex: number, outputOffset: number): number {
+	public getInputOffsetOfOutputPosition(outputLineIndex: number, outputOffset: number): number {
+		let inputOffset = 0;
 		if (outputLineIndex === 0) {
-			return outputOffset;
+			inputOffset = outputOffset;
 		} else {
-			return breakOffsets[outputLineIndex - 1] + outputOffset;
+			inputOffset = this.breakOffsets[outputLineIndex - 1] + outputOffset;
 		}
+
+		if (this.injectionOffsets !== null) {
+			for (let i = 0; i < this.injectionOffsets.length; i++) {
+				if (inputOffset > this.injectionOffsets[i]) {
+					inputOffset -= Math.min(inputOffset - this.injectionOffsets[i], this.injectionTexts![i].length);
```

Comment:
Shoul this be `inputOffset -= this.injectionTexts![i].length;`? What is the extra `Math.min` helpful for?`